trace-mini-agent/src/env_verifier.rs

```diff
@@ -0,0 +1,213 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+use async_trait::async_trait;
+use hyper::{Body, Client, Method, Request, Response};
+use log::{debug, error};
+use serde::{Deserialize, Serialize};
+use std::process;
+
+use datadog_trace_utils::trace_utils;
+
+const GCP_METADATA_URL: &str = "http://metadata.google.internal/computeMetadata/v1/?recursive=true";
+
+#[derive(Default, Debug, Deserialize, Serialize)]
+pub struct GCPMetadata {
+    pub instance: GCPInstance,
+    pub project: GCPProject,
```

Comment:
I think there is also the location/region field in this endpoint (as least this is the case in CloudRun) and we might want this metadata as well for tagging purposes

Reply:
region field is within `GCPInstance`